docs-v2/reference/scripts.mdx

```diff
@@ -99,29 +405,205 @@ await nango.log("This is a log.");
+```js
+await nango.getEnvironmentVariables();
+```
+
+**Parameters**
+
+No parameters.
+
+**Response**
+
+<Expandable>
+```json
+[
+    {
+        "name": "MY_SECRET_KEY",
+        "value": "SK_373892NSHFNCOWFO..."
+    }
+]
+```
+</Expandable>
+
+# Trigger action
 
 Integration scripts currently do not support importing files, which limits the ability to share code between integration scripts. 
 
-As a temporary workaround, you can call action scripts from other integration scripts ([reference](/reference/sdks/node#trigger-an-action)).
+As a temporary workaround, you can call action scripts from other integration scripts with:
+
+```js
+await nango.triggerAction('<ACTION-NAME>', { 'custom_key1': 'custom_value1' });
+```
+
+**Parameters**
+
+<Expandable>
+    <ResponseField name="actionName" type="string" required>
+        The name of the action to trigger.
+    </ResponseField>
+    <ResponseField name="input" type="unkown" required>
+        The necessary input for your action's `runAction` function.
+    </ResponseField>
+</Expandable>
+
+**Response**
+
+<Expandable>
+```json
+{
+    "your-properties": "The data returned by the action"
+}
+```
+</Expandable>
+
+# Paginate through API responses
+
+Follow the pagination [step-by-step guides](/customize/guides/advanced/paginate-api-responses) to learn how to paginate through API responses easily. 
 
-### Paginate through API responses
+# Manage connection metadata
 
-Follow the pagination [step-by-step guides](/customize/guides/advanced/paginate-api-responses) to learn how to easily paginate through API responses. 
+### Get connection metadata
 
-### Manage connection metadata
+Returns the connection's metadata.
 
-[Retrieve](/reference/sdks/node#get-connection-metadata), [edit](/reference/sdks/node#edit-connection-metadata) or [override](/reference/sdks/node#set-connection-metadata) the connection metadata.
+```js
+await nango.getMetadata();
+```
+
+Better, you can specify the type of the metadata;
+
+```ts
+interface CustomMetadata {
+    anyKey: Record<string, string>;
+}
+const myTypedMetadata = await nango.getMetadata<CustomMetadata>();
+```
+
+**Parameters**
+
+No parameters.
+
+**Example Response**
+
+<Expandable>
+```json
+{
+    "custom_key1": "custom_value1"
+}
+```
+</Expandable>
 
-### Get connection
+### Set connection metadata
 
-Get the connection details, including the API credentials ([reference](/reference/sdks/node#get-a-connection-with-credentials)).
+Set custom metadata for the connection (overrides existing metadata).
+
+```js
+await nango.setMetadata({ 'CUSTOM_KEY1': 'CUSTOM_VALUE1' });
+```
+
+**Parameters**
+
+<Expandable>
+    <ResponseField name="metadata" type="Record<string, any>" required>
+        The custom metadata to store in the connection.
+    </ResponseField>
+</Expandable>
+
+**Response**
+
+Empty response.
+
+### Edit connection metadata
+
+Edit custom metadata for the connection. Only overrides specified properties, not the entire metadata.
```

Comment:
Is it clear that it overrides and also adds properties potentially?

```
await nango.updateMetadata({ 
   'CUSTOM_KEY1': 'CUSTOM_VALUE1',
   'new-prop': 'new-value
});

```

Reply:
Edited!